js/private/npm_utils.bzl

```diff
@@ -0,0 +1,47 @@
+"Utility functions for npm rules"
+
+def _bazel_name(name, version):
```

Comment:
This function works as a blacklist of certain symbols that we don't want to be in a bazel name. I think this function should be written to work as a whitelist instead since the number of valid characters that are valid for bazel is very small and well known. 

As one example implementation, you could iterate over the characters of the name. If the character is valid keep it, if it matches a hard coded set of transformations you maintain, transform it, otherwise replace it with a default accepted character